src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -379,6 +379,50 @@ public virtual ReadOnlySpan<byte> Get(ReadOnlySpan<byte> rawKey, Hash256? rootHa
             }
         }
 
+        [DebuggerStepThrough]
+        public byte[]? GetNodeByPath(byte[] nibbles, Hash256? rootHash = null)
+        {
+            try
+            {
+                int nibblesCount = nibbles.Length;
+                CappedArray<byte> result = Run(nibbles, nibblesCount, Array.Empty<byte>(), false, startRootHash: rootHash,
+                    isNodeRead: true);
+                return result.ToArray() ?? Array.Empty<byte>();
+            }
+            catch (TrieException e)
+            {
+                throw new TrieException(
```

Comment:
Move this out to local function; new exception + string interpolation = very bloated stack clearing on method entry even if exception not thrown